src/pybind/mgr/dashboard/frontend/src/app/shared/services/summary.service.spec.ts

```diff
@@ -66,14 +67,80 @@ describe('SummaryService', () => {
     subs.unsubscribe();
   }));
 
+  describe('Should test subscribe without initial value', () => {
+    beforeEach(() => {
+      summaryService['summaryDataSource'].next(undefined);
+    });
+
+    it('should call subscribeOnce', () => {
+      let result;
+      let i = 0;
+      const subscriber = summaryService.subscribeOnce((response) => {
+        i++;
+        result = response;
+      });
+
+      expect(subscriber).toEqual(jasmine.any(Subscriber));
+      expect(i).toBe(0);
+      expect(result).toEqual(undefined);
+
+      summaryService['summaryDataSource'].next(undefined);
```

Comment:
How about using a method that is used here instead?

Like
```ts
  const extendSummaryData = (newData: any) => summaryService['summaryDataSource'].next(newData);
```

Reply:
Done.